torch/optim/optimizer.py

```diff
@@ -790,6 +790,24 @@ def update_group(group: Dict[str, Any], new_group: Dict[str, Any]) -> Dict[str,
             post_hook(self)
 
 
+    def init_state_per_param(self, param: torch.Tensor, param_group: Dict[str, Any]) -> None:
+        r"""Initializes the optimizer state corresponding to the input parameter.
+
+        This method will update the optimizer state in place and will initialize optimizer
```

Comment:
I think this would be a lot clearer with a code sample that shows how a very simple optimizer should implement step.
Also if you explicitly care about mixed dtype, I would give an example of that.